### PR TITLE
Add Visual Studio 2022 Editor Color Scheme

### PR DESCRIPTION
**Pull request by JoeRobich**
Adds new color scheme where the UX team can try new editor colors.

### CODE REVIEW

File: src/VisualStudio/Core/Def/Implementation/ColorSchemes/ColorSchemeApplier.Settings.cs [hunk 1]
```diff
@@ -119,7 +120,7 @@ public void MigrateToColorSchemeSetting()
 
                 var colorScheme = useEnhancedColorsSetting == ColorSchemeOptions.UseEnhancedColors.DoNotUse
                     ? SchemeName.VisualStudio2017
```
[the remark this answers is not in the document]
Comment (JoeRobich) [reply src/VisualStudio/Core/Def/Implementation/ColorSchemes/ColorSchemeApplier.Settings.cs]: Updated the migration method to move VS2019 scheme users to VS2022 scheme

File: src/VisualStudio/Core/Def/Implementation/ColorSchemes/ColorSchemeApplier.cs [hunk 2]
```diff
@@ -122,36 +126,29 @@ private void UpdateColorScheme()
...
+                _settings.ApplyColorScheme(colorScheme, colorSchemeRegistryItems[colorScheme]);
             }
         }
 
         /// <summary>
         /// Returns true if the color scheme needs updating.
         /// </summary>
         /// <param name="colorScheme">The color scheme to update with.</param>
-        private bool TryGetUpdatedColorScheme([NotNullWhen(returnValue: true)] out SchemeName? colorScheme)
+        private bool DoesColorSchemeNeedUpdating(out SchemeName colorScheme)
```
[the remark this answers is not in the document]
Comment (JoeRobich) [reply src/VisualStudio/Core/Def/Implementation/ColorSchemes/ColorSchemeApplier.cs]: I inlined this method.

File: src/VisualStudio/Core/Def/Implementation/ColorSchemes/ColorSchemeApplier.cs [hunk 3]
```diff
@@ -111,7 +112,7 @@ private IVsTask QueueColorSchemeUpdate()
                 VsTaskLibraryHelper.ServiceInstance, VsTaskRunContext.UIThreadBackgroundPriority, () => UpdateColorScheme());
         }
 
-        private void UpdateColorScheme()
+        private void UpdateColorScheme(bool forceApply = false)
```
[the remark this answers is not in the document]
Comment (JoeRobich) [reply src/VisualStudio/Core/Def/Implementation/ColorSchemes/ColorSchemeApplier.cs]: Verified that the force apply was unnecessary when F5-ing. Reverted.

File: src/VisualStudio/Core/Def/Implementation/ColorSchemes/ColorSchemeApplier.cs [hunk 4]
```diff
@@ -92,7 +92,8 @@ public void Initialize()
             // with the Roslyn colors before they are seen by the user. This is important because the MEF exported Roslyn classification
             // colors are only applicable to the Blue and Light VS themes.
 
-            UpdateColorScheme();
+            // TODO: Revert before merging to main. This change was made so that we could force reapply the color scheme on start up.
+            UpdateColorScheme(forceApply: true);
```
[the remark this answers is not in the document]
Comment (JoeRobich) [reply src/VisualStudio/Core/Def/Implementation/ColorSchemes/ColorSchemeApplier.cs]: Verified the force apply was unnecessary. reverted

File: src/EditorFeatures/Core/Options/ColorSchemeOptions.cs [hunk 5]
```diff
@@ -39,7 +39,8 @@ public ColorSchemeOptions()
 
         public enum UseEnhancedColors
         {
-            Migrated = -2,
+            MigratedFromVS2019ColorScheme = -3,
```
Comment (JoeRobich) [inline src/EditorFeatures/Core/Options/ColorSchemeOptions.cs:42]: Added new one-time migration to move from the VS2019 scheme.

**Comment by JoeRobich** [comment]
@chryw Sorry for the delay in getting this merged. Will prioritize getting it in.

**Comment by JoeRobich** [comment]
@chryw The features/vs2022-theme branch is now ready for you to work against. Please don't push changes to the branch, but instead open a PR when you are ready to make updates. Thanks!

**Comment by chryw** [comment]
Thanks @JoeRobich ! I'll use a fork and PR from there. :-)

**Pull request merged**